### PR TITLE
Fix: Amp experiment variable access in shadow-v0.js

### PR DESCRIPTION
**Pull request by prateekbh**
- access body through ampdoc

fixes https://github.com/ampproject/amphtml/issues/19434

### CODE REVIEW

File: extensions/amp-experiment/0.1/test/test-amp-experiment.js [hunk 1]
```diff
@@ -145,4 +145,13 @@ describes.realWin('amp-experiment', {
           .to.equal(null);
     });
   });
+
+  it('should access document from ampdoc', () => {
+    addConfigElement('script');
```
Comment (lannka) [inline extensions/amp-experiment/0.1/test/test-amp-experiment.js]: I understand this is difficult to do unit test here. But I don't think this test is adding value other than making sure that line of code is not touched.

Sounds like calling `this.win.document` and `doc.body` is in general a dangerous thing, should we add a presubmit check? @choumx 

I suggest you revert this test case, and we can have that check in a separate PR.

Comment (choumx) [reply extensions/amp-experiment/0.1/test/test-amp-experiment.js]: A presubmit check for code in `extensions/` might be a good idea.

Comment (prateekbh) [reply extensions/amp-experiment/0.1/test/test-amp-experiment.js]: reverted

Comment (kristoferbaxter) [reply extensions/amp-experiment/0.1/test/test-amp-experiment.js]: @choumx Should @prateekbh add this presubmit check or would you prefer it in a separate PR?

Comment (choumx) [reply extensions/amp-experiment/0.1/test/test-amp-experiment.js]: Separately is fine. Filed #19514.

**Comment by choumx** [comment]
Can we have a better PR title? These will be visible in our release notes. :)

File: extensions/amp-experiment/0.1/amp-experiment.js [hunk 2]
```diff
@@ -81,11 +80,11 @@ export class AmpExperiment extends AMP.BaseElement {
    * @private
    */
   addToBody_(experiments) {
-    const doc = this.win.document;
-    return waitForBodyPromise(doc).then(() => {
+    const doc = this.getAmpDoc();
+    return doc.whenBodyAvailable().then(() => {
```
Comment (choumx) [inline extensions/amp-experiment/0.1/amp-experiment.js]: This promise is resolved with the body.

```js
return doc.whenBodyAvailable().then(body => {
  // ...
  body.setAttribute(...);
});
```

File: extensions/amp-experiment/0.1/test/test-amp-experiment.js (hunk 1 above)
Comment (choumx) [reply extensions/amp-experiment/0.1/test/test-amp-experiment.js]: We should have a PWA integration test for each AMP component, including this one.

File: extensions/amp-experiment/0.1/amp-experiment.js (hunk 2 above)
Comment (prateekbh) [reply extensions/amp-experiment/0.1/amp-experiment.js]: TIL

**Comment by prateekbh** [comment]
sry.. new machine mess up..

**Pull request merged**